packages/api-server/src/convert-tx.ts

```diff
@@ -174,15 +213,32 @@ async function parseRawTransactionData(
 
   const publicKey = recoverPublicKey(signature, message);
   const fromEthAddress = publicKeyToEthAddress(publicKey);
-  const fromId: HexNumber | undefined = await getAccountIdByEthAddress(
+  let fromId: HexNumber | undefined = await getAccountIdByEthAddress(
     fromEthAddress,
     rpc
   );
 
+  let cacheKeyAndValue: [string, string] | undefined;
   if (fromId == null) {
-    throw new Error(
-      `from id not found by from Address: ${fromEthAddress}, have you deposited?`
+    const ethTxHash = calcEthTxHash(rlpEncoded);
+
+    cacheKeyAndValue = await cacheAutoCreateAccount(
+      polyjuiceRawTx,
+      ethTxHash,
+      fromEthAddress,
+      rpc,
+      gasLimit,
+      gasPrice,
+      value
     );
+
+    if (!cacheKeyAndValue) {
+      throw new Error(
+        `from id not found by from Address: ${fromEthAddress}, have you deposited?`
+      );
+    }
+
+    fromId = "0x0";
```

Comment:
should we extract this special "0x0" to a const, like AUTO_CREATE_FROM_ID or something else?

Reply:
Updated to use `AUTO_CREATE_ACCOUNT_FROM_ID`